### PR TITLE
[SPARK-6800][SQL] Update doc for JDBCRelation's columnPartition

### PR DESCRIPTION
**Pull request by viirya**
JIRA https://issues.apache.org/jira/browse/SPARK-6800

### CODE REVIEW

**Comment by marmbrus** [comment]
This is not a bug, this is [the intended behavior](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala#L57).  I think the right fix here is to update the programming guide and other documentation to clearly state this.  If users want to limit the scope of the data, they should add predicates which will be pushed down to the database.

**Comment by viirya** [comment]
ok. If I understand it correctly, `lowerBound` and `upperBound` are just used to decide partition stride, not for filtering. So all table rows are partitioned.

**Comment by marmbrus** [comment]
That is correct.

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala [hunk 1]
```diff
@@ -50,9 +50,11 @@ private[sql] object JDBCRelation {
    * Given a partitioning schematic (a column of integral type, a number of
    * partitions, and upper and lower bounds on the column's value), generate
    * WHERE clauses for each partition so that each row in the table appears
-   * exactly once.  The parameters minValue and maxValue are advisory in that
+   * exactly once. The parameters minValue and maxValue are advisory in that
    * incorrect values may cause the partitioning to be poor, but no data
-   * will fail to be represented.
+   * will fail to be represented. Note: the upper and lower bounds are just
+   * used to decide partition stride, not for filtering. So all the rows in
+   * table will be partitioned.
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: > The parameters minValue and maxValue are advisory in that incorrect values may cause the partitioning to be poor, but no data will fail to be represented.

The sentence above already explains that the filters are only used for partitioning and that all data will always be returned.  I think the best place to update would be in the [SQL programming guide](https://github.com/apache/spark/blob/master/docs/sql-programming-guide.md), in the table under the section "JDBC To Other Databases".

Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JDBCRelation.scala]: Updated.

**Comment by micaelcapitao** [comment]
Hi.
The doc for the SQLContext.jdbc() method says:
> @param columnName the name of a column of integral type that will be used for partitioning.
> @param lowerBound the minimum value of `columnName` to retrieve
> @param upperBound the maximum value of `columnName` to retrieve
> @param numPartitions the number of partitions.  the range `minValue`-`maxValue` will be split
>                       evenly into this many partitions

This doc should be updated too.

How can one add predicates to limit the scope of data being pushed from the DB using the SQLContext API?

**Comment by viirya** [comment]
@micaelcapitao Thanks. I updated the doc too.

I think you can use jdbc data source API to create temporary table and then use WHERE clause to add predicates.

**Comment by marmbrus** [comment]
That's correct. WHERE clause predicates and data frame filter operations
will be pushed down to the database.

**Comment by marmbrus** [comment]
Thanks!  Merged to master and branch-1.3

**Pull request closed without merging**